src/daemon/daemon.cpp

```diff
@@ -1766,11 +1747,25 @@ try // clang-format on
                     vm.get_snapshot(snapshot); // verify validity of any snapshot name requested separately
 
                 for (const auto& snapshot : vm.view_snapshots())
-                    get_snapshot_info(snapshot);
+                {
+                    auto overview = response.mutable_snapshot_overview()->add_overview();
+                    auto fundamentals = overview->mutable_fundamentals();
+
+                    overview->set_instance_name(name);
+                    populate_snapshot_fundamentals(snapshot, fundamentals);
+                }
             }
             else
+            {
                 for (const auto& snapshot : pick)
-                    get_snapshot_info(vm.get_snapshot(snapshot));
+                {
+                    auto overview = response.mutable_snapshot_overview()->add_overview();
+                    auto fundamentals = overview->mutable_fundamentals();
+
+                    overview->set_instance_name(name);
```

Comment:
Why didn't you include these lines in `populate_snapshot_fundamentals`, instead of repeating them?

Reply:
I wanted to keep `populate_snapshot_fundamentals` aligned with the message type `SnapshotFundamentals` and not change anything outside of it. There is also [one situation](https://github.com/canonical/multipass/blob/ac8a96cb4d9819136f68f428151cdeb0f1359154/src/daemon/daemon.cpp#L1327) where the fundamentals are set and not `instance_name`.

Reply:
I see. But then you can extract a helper employ here, similar to what happened already with `get_snapshot_info`.